include/arrayfire.h

```diff
@@ -97,6 +97,80 @@
       diff, gradient, etc.
    @}
 
+   @defgroup memory_manager Memory Management
+   @{
+
+      \brief Interfaces for writing custom memory managers.
+
+      Create and set a custom memory manager by first defining the relevant
+closures for each required function, for example:
+
+      \code{.cpp}
+      af_err initialize(af_memory_manager manager) {
```

Comment:
perhaps use the name my_initialize instead of initialize to specify that this is not something in ArrayFire but a user defined function.